src/core/reference/include/openvino/reference/fake_quantize.hpp

```diff
@@ -21,31 +21,86 @@ namespace ov {
 namespace reference {
 namespace fake_quantize_details {
 template <typename T>
-inline T quantize(const T& arg,
-                  const T& in_low,
-                  const T& in_high,
-                  const T& out_low,
-                  const T& out_high,
-                  const size_t& levels) {
+static inline T quantize(const T arg,
+                         const T in_low,
+                         const T in_high,
+                         const T out_low,
+                         const T out_high,
+                         const T levels_minus_one) {
     if (arg <= std::min(in_low, in_high)) {
         return out_low;
     } else if (arg > std::max(in_low, in_high)) {
         return out_high;
     }
-    return static_cast<T>(std::nearbyint((arg - in_low) / (in_high - in_low) * (levels - 1)) / (levels - 1) *
+    return static_cast<T>(std::nearbyint((arg - in_low) / (in_high - in_low) * levels_minus_one) / levels_minus_one *
                               (out_high - out_low) +
                           out_low);
 }
 
+static std::vector<size_t> compute_strides(const ov::Shape& out_shape, const ov::Shape& shape);
```

Comment:
could you describe the approach in general and what "stride", "inner_stride" and "outer_stride" mean?

Reply:
done, let me know if this needs to be expanded